### ISSUE TITLE
Some files with rests removed from the sub voice are recognized as corrupted

### ISSUE THREAD

**Issue by bakajikara** [musescore/MuseScore#16841]
### Issue type

File corruption

### Steps to reproduce

1. input this
![image](https://user-images.githubusercontent.com/102702868/225331852-b2a7cd0c-48d8-4690-8bbd-24906a817750.png)

2. delete first two rests in voice 2
![image](https://user-images.githubusercontent.com/102702868/225331870-25e00976-2c2f-4a77-a0c4-6dca5ec86901.png)

3. save score
4. load score
5. file corrupted
![image](https://user-images.githubusercontent.com/102702868/225331899-481b8531-80c2-4d45-b4d8-00cd763b7fbe.png)

### MuseScore Version

Latest 4.1 Nightly build (OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.1.0-230740305, revision: github-musescore-musescore-05cb052)

### Regression

Yes, this used to work in a previous version of Musescore 4.x

### Operating system

Windows 11

### Additional context

- Does not occur in the March 14 nightly build and earlier.
- This seems to happen when 5 eighth rests are deleted.
- This also occurs with files saved in other builds.

**Comment by cbjeukendrup** [comment]
@mike-spa I believe you worked on something with rests recently, might it be possible that this is related to that?

**Comment by oktophonie** [comment]
(That's https://github.com/musescore/MuseScore/pull/16260)

**Comment by mike-spa** [comment]
@cbjeukendrup No, the only work I did on rests is in the PR that Simon linked, and I don't think it is related (but we could try to revert it and see if indeed it is). 
What's happening here is that, when rests in the second voice are deleted, they are replaced by a "special" rest called "gap". It's treated as a rest internally, but it has no graphical output. The whole idea always seemed a bit silly to me, cause everything you can do with "gaps" you can do more easily (and more _correctly_) with invisible rests. Gaps are basically just "more invisible", and they cause a lot of different troubles, also for horizontal spacing. I suspect the corruption checks have some problem in interpreting these gaps (or the gaps themselves have problems). I didn't change anything related to gaps so far, except for details about how to handle them in spacing. I'd be glad to see them gone forever, tbh, unless they have important use cases that I'm just missing.

**Comment by MarcSabatella** [comment]
There has always been weirdness associated with removing rests.  "Gap" rests are a pretty new thing, I gather; in the past, removing a rest simply left a hole, which created its own problems.  I'm not clear at all on why gap rests were introduced.

FWIW, I have long advocated we consider getting rid of the ability to delete rests at all - I agree that simply making them invisible simplifies life both for the user and for the code.  And there was a time when we tried making delete of a rest automatically just turn it invisible.  This turned out to not be popular at all, though, so it was reverted.  Maybe that's when the gap rests were created?

**Comment by mike-spa** [comment]
So, it turns out that this issue is exactly the same issue as #16260, except this one you don't see because it involves gap rests (i.e. invisible rests generated internally to fill the voice). See also Marc's [comment](https://github.com/musescore/MuseScore/pull/16260#discussion_r1140461358). When I fixed that one I excluded gap rests cause I thought they followed different rules, but apparently they don't. Fix coming soon.
(As in: new scores won't have this kind of corruption again. Corrupted scores will remain corrupted. They can be fixed by deleting and rewriting the interested measure).
